### PR TITLE
User Service MVP

### PR DESCRIPTION
**Pull request by ianaspivac**
- Added user service proto/implementation
- Simple JWT token construction/verification
- Password hashing

### CODE REVIEW

File: services/user/Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	--go-grpc_out=. --go-grpc_opt=paths=source_relative \
+	./services/user/user.proto
+
+run:
+	go run cmd/user.go
+
+run-db:
+	docker compose up --build -d db
+
+run-local: run-db run
```
Comment (FoxFurry) [inline services/user/Makefile]: Please add newline at the end of file

Comment (ianaspivac) [reply services/user/Makefile]: Done

File: services/user/migrations/credentials_table.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE credentials (
+    id INT UNSIGNED PRIMARY KEY AUTO_INCREMENT NOT NULL,
+    name VARCHAR(255) NOT NULL,
+    password VARCHAR(255) NOT NULL
+) DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
```
Comment (FoxFurry) [inline services/user/migrations/credentials_table.sql]: Added here newline as well

Comment (ianaspivac) [reply services/user/migrations/credentials_table.sql]: Done!

**Review by FoxFurry — APPROVED** [review]
Looks good! :shipit:

**Pull request merged**